app/models/pipeline_run.rb

```diff
@@ -1390,4 +1391,35 @@ def outputs_by_step(can_see_stage1_results = false)
   def self.viewable(user)
     where(sample_id: Sample.viewable(user).pluck(:id))
   end
+
+  # Keys here are used as cache keys for report_info action in SamplesController.
+  # The values here are used as defaults for PipelineSampleReport.jsx.
+  def report_info_params
+    {
+      pipeline_version: pipeline_version || PipelineRun::PIPELINE_VERSION_WHEN_NULL,
+      # TODO: (gdingle): is this good enough for cache hits? see get_background_id
```

Comment:
Actually background_id selection is persisted in a cookie. So all reports will be opened with the background that the user last selected. `get_background_id` just checks if the user is allowed to access the selected background (if not, it uses default public background instead).

Reply:
Ok so what would you recommend given the expected access pattern? I noticed this line
```
viewable_background_ids = current_power.backgrounds.pluck(:id)
```
We could simply cache a report for all `viewable_background_ids`.

Reply:
Yes. Would you use the `Power` of the user that uploaded the sample? What about other users who have access by way of project membership? I think we could precache for *all* background_ids, but need to bear in mind we will need to enforce access control at request time

Reply:
Ok sounds good. I'll see about access control at request time.